src/gate_detect/src/GateDetector.cpp

```diff
@@ -263,140 +224,99 @@ std::vector<Pole> GateDetector::findHorPoles(std::vector<cv::Vec4i> horLines) {
+            gate.topPole = *std::min_element(
             horPoles.begin(), horPoles.end(), [](Pole lhs, Pole rhs) {
                 return lhs.getHorMid() < rhs.getHorMid();
             });
 
-            if (leftPole.getVertMid() > topPole.getVertMid()) {
-                gateCoordinates.detectedLeftPole = 0;
-                gateCoordinates.angleLeftPole    = 0;
-                gateCoordinates.distanceLeftPole = 0;
-
-                gateCoordinates.detectedRightPole = 1;
-                gateCoordinates.angleRightPole =
-                rightPole.getVertAngle(imagePixelWidth_);
-                gateCoordinates.distanceRightPole = rightPole.getVertDistance();
-
-                gateCoordinates.detectedTopPole = 1;
-                gateCoordinates.angleTopPole =
-                topPole.getHorAngle(imagePixelHeight_);
-                gateCoordinates.distanceTopPole = topPole.getHorDistance();
-
-            } else if (rightPole.getVertMid() < topPole.getVertMid()) {
-                gateCoordinates.detectedLeftPole = 1;
-                gateCoordinates.angleLeftPole =
-                leftPole.getVertAngle(imagePixelWidth_);
-                gateCoordinates.distanceLeftPole = leftPole.getVertDistance();
-
-                gateCoordinates.detectedRightPole = 0;
-                gateCoordinates.angleRightPole    = 0;
-                gateCoordinates.distanceRightPole = 0;
-
-                gateCoordinates.detectedTopPole = 1;
-                gateCoordinates.angleTopPole =
-                topPole.getHorAngle(imagePixelHeight_);
-                gateCoordinates.distanceTopPole = topPole.getHorDistance();
-            } else {
-                gateCoordinates.detectedLeftPole = 1;
-                gateCoordinates.angleLeftPole =
-                leftPole.getVertAngle(imagePixelWidth_);
-                gateCoordinates.distanceLeftPole = leftPole.getVertDistance();
-
-                gateCoordinates.detectedRightPole = 1;
-                gateCoordinates.angleRightPole =
-                rightPole.getVertAngle(imagePixelWidth_);
-                gateCoordinates.distanceRightPole = rightPole.getVertDistance();
-
-                gateCoordinates.detectedTopPole = 1;
-                gateCoordinates.angleTopPole =
-                topPole.getHorAngle(imagePixelHeight_);
-                gateCoordinates.distanceTopPole = topPole.getHorDistance();
+            if (gate.leftPole.getVertMid() > gate.topPole.getVertMid()) {
+                gate.leftDetected = false;
+
+                gate.rightDetected = true;
+
+                gate.topDetected = true;
+
+            }
+
+            else if (gate.rightPole.getVertMid() < gate.topPole.getVertMid()) {
+                gate.leftDetected = true;
+
+                gate.rightDetected = false;
+
+                gate.topDetected = true;
+
+            }
+
+            else {
+                gate.leftDetected = true;
+
+                gate.rightDetected = true;
+
```

Comment:
I would probably suggest removing all the extra spaces here, I don't think they help readability much, and in fact probably impair it a bit just because one has to scroll to see the entire function

Reply:
Done